src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java

```diff
@@ -271,6 +271,38 @@ public void onInstallShiny(InstallShinyEvent event)
                 new Command() { public void execute() {}});
    }
    
+   public void withDataImport(String userAction, final Command command)
+   {
+     withDependencies(
+        "Preparing Data Import",
+        userAction, 
+        dataImportDependenciesArray(), 
+        false,
+        new CommandWithArg<Boolean>()
+        {
+           @Override
+           public void execute(Boolean succeeded)
+           {
+              if (succeeded)
+                 command.execute();
+           }
+        }
+     );
+   }
+   
+   private ArrayList<Dependency> dataImportDependencies()
+   {
+      ArrayList<Dependency> deps = new ArrayList<Dependency>();
+      deps.add(Dependency.cranPackage("readr", "0.2.2"));
```

Comment:
For sub-dependencies that have version requirements it's good to add those as well (e.g. Rcpp 0.11.5 as seen here: https://cran.r-project.org/web/packages/readr/index.html).